Coding/fw/lib/WebServer/ESP_Server.cpp

```diff
@@ -0,0 +1,251 @@
+
+/* PROTOTYPES *************************************************************************************/
+/**
+*   Determines the state of the WiFiModeSelect Button to enter AP Mode
+*   @return bool AP Mode. If true, will request the AP mode. If false, requests STA Mode
+*/
+static bool setAPMode();
+
+/**
+*  Variable to call Restart
+*/
+static bool restartRequested = false;
+
+/**
+* Connects to the WiFi AP when requested
+* @return true when succesfully connected. False otherwise.
+*/
+static bool connectWiFi();
+
+/**
+* Registers the handlers on the server, depending on the WiFi Mode chosen
+* @param bool apModeRequested  determines if AP Mode or STA Mode are requested, to choose the correct handlers
+* return success
+*/
+static bool initPages(bool apModeRequested);
+
+/* VARIABLES **************************************************************************************/
+IPAddress m_serverIP; /**< Stores the IP Address of the ESP32 */
+
+/* PUBLIC METHODES ********************************************************************************/
+
+/**************************************************************************************************/
+
+/**
+*   Initializing of AsyncWebServer, DNS and WiFi capabilities. 
+*   return success
+*/
+bool ESPServer::begin()
+{
+    bool success = true;
+    WebConfig::importConfig(); /*< Imports Credentials from Flash Memory */
+
+    if (setAPMode())
+    {
+        if (!WiFi.softAP(WebConfig::getAP_SSID().c_str(), WebConfig::getAP_PASS().c_str()))
+        {
+            success = false;
+        }
+
+        m_serverIP = WiFi.softAPIP();
+
+        if (!initPages(true))
+        {
+            success = false;
+        }
+    }
+    else
+    {
+        if (!WiFi.mode(WIFI_STA))
+        {
+            success = false;
+        }
+        else if (WiFi.begin(WebConfig::getSTA_SSID().c_str(), WebConfig::getSTA_PASS().c_str()) == WL_CONNECT_FAILED)
+        {
+            success = false;
+        }
+        else
+        {
+            success = connectWiFi();
+        }
+
+        if (!initPages(false))
+        {
+            success = false;
+        }
+    }
+
+    dnsServer.setErrorReplyCode(DNSReplyCode::NoError);
+
+    if (!dnsServer.start(WebConfig::DNS_PORT, "*", m_serverIP))
+    {
+        success = false;
+    }
+    else if (!SPIFFS.begin())
+    {
+        success = false;
+    }
+
+    Serial.println(m_serverIP);
+
+    server.begin();
+    return success;
+}
+
+/**************************************************************************************************/
+
+/**
+*   Disconnects and disables the WebServer
+*/
+bool ESPServer::end()
+{
+    server.end();
+    SPIFFS.end();
+    dnsServer.stop();
+    return WiFi.disconnect(true, true);
+}
+
+/**************************************************************************************************/
+/**
+*   Calls next request on DNS Server
+*   return true
+*/
+bool ESPServer::handleNextRequest()
```

Comment:
the return comment is a lying. 
the combination of handleNext request and restartRequested combines unrelated things. Think about a better solution to signal a restart need.